### PR TITLE
doc: translate State as a Snapshot

### PR DESCRIPTION
**Pull request by Annie-Chien**
I've finished translating "State as a Snapshot". 
It's my first time making a contribution, please let me know if there is anything that can be improved.

Thank you so much!

### CODE REVIEW

**Comment by neighborhood999** [comment]
> I've finished translating "State as a Snapshot". It's my first time making a contribution, please let me know if there is anything that can be improved.
> 
> Thank you so much!

Hi @Annie-Chien , thank you contribution, can you help to resolve conflicts?

**Comment by Annie-Chien** [comment]
Done. Thank you!

File: src/content/learn/state-as-a-snapshot.md [hunk 1]
```diff
@@ -139,18 +136,20 @@ Notice that `number` only increments once per click!
...
-    - React prepares to change `number` to `1` on the next render.
-3. `setNumber(number + 1)`: `number` is `0` so `setNumber(0 + 1)`.
-    - React prepares to change `number` to `1` on the next render.
 
-Even though you called `setNumber(number + 1)` three times, in *this render's* event handler `number` is always `0`, so you set the state to `1` three times. This is why, after your event handler finishes, React re-renders the component with `number` equal to `1` rather than `3`.
+雖然呼叫了 `setNumber(number + 1)` 三次，在*這一次 render 的* event handler 內的 `number` 一直都是 `0`，所以等同於你把 state 設定為 `1` 三次。這就是為什麼在 event handler 執行結束後，React 用等於 `1` 而非 `3` 的 `number` 來重新 render component。
 
-You can also visualize this by mentally substituting state variables with their values in your code. Since the `number` state variable is `0` for *this render*, its event handler looks like this:
+你也可以透過在心裡將程式碼中的 state 變數替換為它們的值來視覺化這一切。
+由於在*這一次 render* 中，state 變數 `number` 的值為 `0`，它的 event handler 看起來就像是這樣：
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:152]: 跟著上移行，不需要分行

File: src/content/learn/state-as-a-snapshot.md [hunk 2]
```diff
@@ -170,11 +169,10 @@ For the next render, `number` is `1`, so *that render's* click handler looks lik
 }}>+3</button>
 ```
 
-This is why clicking the button again will set the counter to `2`, then to `3` on the next click, and so on.
+這就是為什麼再次點擊按鈕會將計數器設置為 `2`，然後在下一次點擊時會設置為 `3`，依此類推。
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:172]:
```suggestion
這就是為什麼再次點擊按鈕會將計數器設定為 `2`，然後在下一次點擊時會設定為 `3`，依此類推。
```

File: src/content/learn/state-as-a-snapshot.md [hunk 3]
```diff
@@ -170,11 +169,10 @@ For the next render, `number` is `1`, so *that render's* click handler looks lik
...
 
-## State over time {/*state-over-time*/}
-
-Well, that was fun. Try to guess what clicking this button will alert:
+## 隨著時間改變的 state {/*state-over-time*/}
+嗯，那真是有趣。試著猜猜看點擊這個按鈕會彈出什麼提示框：
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:175]: 需要空一行

File: src/content/learn/state-as-a-snapshot.md [hunk 4]
```diff
@@ -203,14 +201,15 @@ h1 { display: inline-block; margin: 10px; width: 30px; text-align: center; }
...
+如果你使用之前提到的替換法，你可以猜到提示框會顯示「0」：
+
 
 ```js
 setNumber(0 + 5);
 alert(0);
 ```
 
-But what if you put a timer on the alert, so it only fires _after_ the component re-rendered? Would it say "0" or "5"? Have a guess!
+但要是你在提示框上設置計時器，使其在 component 重新 render _之後_ 才觸發呢？那麼它會顯示「5」還是「0」？猜猜看！
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:212]:
```suggestion
但要是你在提示框上設定計時器，使其在 component 重新 render _之後_ 才觸發呢？那麼它會顯示「5」還是「0」？猜猜看！
```

File: src/content/learn/state-as-a-snapshot.md [hunk 5]
```diff
@@ -305,29 +303,25 @@ label, textarea { margin-bottom: 10px; display: block; }
 
 </Sandpack>
 
-**React keeps the state values "fixed" within one render's event handlers.** You don't need to worry whether the state has changed while the code is running.
-
-But what if you wanted to read the latest state before a re-render? You'll want to use a [state updater function](/learn/queueing-a-series-of-state-updates), covered on the next page!
-
+**React 會使 state 值在同一次 render 內的 event handler 保持「固定不變」。** 你不需要擔心 state 在程式碼執行時有所異動。
+但要是你希望在重新 render 之前讀取最新的 state 呢？你將會需要使用 [state 的更新函數](/learn/queueing-a-series-of-state-updates)，這會下一章節中介紹！
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:307]:
```suggestion
但要是你希望在重新 render 之前讀取最新的 state 呢？你將會需要使用 [state 的更新函式](/learn/queueing-a-series-of-state-updates)，這會下一章節中介紹！
```

File: src/content/learn/state-as-a-snapshot.md [hunk 6]
```diff
@@ -305,29 +303,25 @@ label, textarea { margin-bottom: 10px; display: block; }
...
-* Every render (and functions inside it) will always "see" the snapshot of the state that React gave to *that* render.
-* You can mentally substitute state in event handlers, similarly to how you think about the rendered JSX.
-* Event handlers created in the past have the state values from the render in which they were created.
+* 設定 state 會請求一次新的 render。
+* React 將 state 儲存在 component 外，好比在架子上一樣。
+* 當你呼叫 `useState`，React 會*為該次 render* 拍一張 state 的快照。
+* 變數和 event handler 不會在重新 render 時「存活」。每次 render 都有自己的 event handler。
+* 每次 render（和其內部的函式）始終會「看到」React 為*該次* render 所提供的 state 快照。
+* 你可以在內心替换 event handler 中的 state，類似於替換被 render 的 JSX。
+* 過去創建的 event handler 保有它們被創建的那一次 render 中的 state 值。
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:315]:
```suggestion
* 過去建立的 event handler 保有它們被建立的那一次 render 中的 state 值。
```

File: src/content/learn/state-as-a-snapshot.md [hunk 7]
```diff
@@ -362,14 +356,13 @@ h1 { margin-top: 20px; }
 
 </Sandpack>
 
-Add an `alert` to the click handler. When the light is green and says "Walk", clicking the button should say "Stop is next". When the light is red and says "Stop", clicking the button should say "Walk is next".
+請在 click handler 裡添加一個 `alert`。當燈是綠色的並顯示「Walk」時，點擊按鈕應顯示「Stop is next」。當燈是紅色的並顯示「Stop」時，點擊按鈕應顯示「Walk is next」。
```
Comment (neighborhood999) [inline src/content/learn/state-as-a-snapshot.md:359]:
```suggestion
請在 click handler 裡加入一個 `alert`。當燈是綠色的並顯示「Walk」時，點擊按鈕應顯示「Stop is next」。當燈是紅色的並顯示「Stop」時，點擊按鈕應顯示「Walk is next」。
```

**Review by neighborhood999 — COMMENTED** [review]
Hi @Annie-Chien I gave you some suggestions, please take a look and feel free to leave comments if you have any questions.

By the way, some paragraph numbers are wrong , please check the original document and made them consistent.

**Review by neighborhood999 — COMMENTED** [review]
Thank you!

**Pull request merged**